docs/api-markdown-documenter/build-api-nav.js

```diff
@@ -0,0 +1,76 @@
+const { ApiItemKind, ApiItemUtilities } = require("@fluid-tools/api-markdown-documenter");
+
+const fs = require("fs-extra");
+const path = require("path");
+const yaml = require("js-yaml");
+
+/**
+ * Processes documents and generates data required for the nav bar.
+ * @param {Array<Object>} documents - List of documents with apiItem.
+ * @param {ApiItem | undefined} documents.apiItem - The API item that the document is created from. Some documents may not have an apiItem.
+ */
+async function buildNavBar(documents) {
+	const validKinds = new Set([
+		ApiItemKind.Class,
+		ApiItemKind.Interface,
+		ApiItemKind.Enum,
+		ApiItemKind.Namespace,
+	]);
+	const { allAPIs, packageMap } = documents.reduce(
+		({ allAPIs, packageMap }, { apiItem }) => {
+			if (apiItem === undefined) {
+				return { allAPIs, packageMap };
+			}
+
+			const { displayName, kind } = apiItem;
+
+			const associatedPackage = apiItem.getAssociatedPackage();
+			const packageName =
+				associatedPackage === undefined
+					? undefined
+					: ApiItemUtilities.getUnscopedPackageName(associatedPackage);
+
+			if (kind === ApiItemKind.Package) {
+				if (packageMap.hasOwnProperty(displayName)) {
+					throw new Error("Package name collision!");
+				}
+
+				packageMap[displayName] = packageName;
+			} else if (validKinds.has(kind)) {
+				allAPIs[packageName] = allAPIs[packageName] || {};
+				allAPIs[packageName][kind] = allAPIs[packageName][kind] || [];
+				allAPIs[packageName][kind].push(displayName);
+			}
+
+			return { allAPIs, packageMap };
+		},
+		{ allAPIs: {}, packageMap: {} },
+	);
+
+	const results = await Promise.allSettled([
+		saveToFile("allAPIs.yaml", allAPIs),
+		saveToFile("packageNameToDisplayName.yaml", packageMap),
+		saveToFile("displayNameToPackageName.yaml", invertMap(packageMap)),
+	]);
+
+	results.forEach((result, index) => {
```

Comment:
Should we be effectively catching the errors here? If these files aren't generated correctly, subsequent build steps won't work. Seems like it might be better to just do `return Promise.all(...)` and let the consuming script exit the process with an error code?